runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleBuilders.kt

```diff
@@ -166,10 +180,16 @@ public class SerializersModuleBuilder internal constructor() : SerialModuleColle
         // Check for overwrite
         val name = concreteSerializer.descriptor.serialName
         val baseClassSerializers = polyBase2Serializers.getOrPut(baseClass, ::hashMapOf)
-        if (!allowOverwrite && concreteClass in baseClassSerializers) throw SerializerAlreadyRegisteredException(
-            baseClass,
-            concreteClass
-        )
+        if (!allowOverwrite) {
+            val previous = baseClassSerializers[concreteClass]
+            if (previous != null) {
+                if (previous == concreteSerializer) return
+                throw SerializerAlreadyRegisteredException(
+                    "Polymorphic serializer for $concreteClass already registered in this module: $previous, " +
```

Comment:
You've decided not to extract `serialName`s from descriptors in this message. Why?

Reply:
Added it back to be consistent with the rest of the changes